plugins/zns/zns.c

```diff
@@ -627,23 +637,54 @@ static int report_zones(int argc, char **argv, struct command *cmd, struct plugi
 		}
 	}
 
-	if (cfg.num_descs == -1) {
-		struct nvme_zone_report r;
+	err = nvme_identify_ns(fd, cfg.namespace_id, false, &id_ns);
+	if (err) {
+		nvme_show_status(err);
+		goto close_fd;
+	}
 
-		err = nvme_zns_report_zones(fd, cfg.namespace_id, 0,
-			0, cfg.state, 0, sizeof(r), &r);
-		if (err > 0) {
-			nvme_show_status(err);
-			goto close_fd;
-		} else if (err < 0) {
-			perror("zns report-zones");
-			goto close_fd;
-		}
-		cfg.num_descs = le64_to_cpu(r.nr_zones);
+	err = nvme_zns_identify_ns(fd, cfg.namespace_id, &id_zns);
+	if (!err) {
+		/* get zsze field from zns id ns data - needed for offset calculation */
+		lbaf = id_ns.flbas & NVME_NS_FLBAS_LBA_MASK;
+	    zsze = le64_to_cpu(id_zns.lbafe[lbaf].zsze);
+	}
+	else {
+		nvme_show_status(err);
+		goto close_fd;
 	}
 
-	report_size = sizeof(struct nvme_zone_report) + cfg.num_descs *
-		(sizeof(struct nvme_zns_desc) + cfg.num_descs * zdes);
+	log_len = sizeof(struct nvme_zone_report);
+	struct nvme_zone_report *buff;
```

Comment:
Please don't mix declarations in the code.

Reply:
Fixed in latest commit.